### PR TITLE
Add service client to get service name

### PR DESCRIPTION
**Pull request by doncem**
[JIRA](https://tools.hmcts.net/jira/browse/RPE-176)

Simple service name getter

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.authorisation;
+
+import feign.FeignException;
+import uk.gov.hmcts.reform.authorisation.exceptions.AbstractAuthorisationException;
+
+public class ServiceDetailClient {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: perhaps we could just add an extra method to `ServiceAuthTokenValidator`?

**Comment by rkondratowicz** [comment]
bump version in gradle.

File: src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: i was afraid if i do, some one would have commented "it does not represent any validation" or something like that

on the other hand, client can incorporate multiple interfaces if we want to and really just use client in a project but specific interface in relevant place. just a thought

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '0.1.0'
+version '0.1.1'
```
Comment (rlewan) [inline build.gradle]: Would you mind changing the version to `0.2.0`? This change looks like neither a bugfix nor a minor modification.

File: src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: it actually does and returns service name 'as a bonus'

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: it does, but does not represent as in endpoint definition: just looking at `GET /details` does imply validation

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: poor route naming in s2s doesn't have to have an impact on functions in this java library imho :)

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java]: _fairs enoughs_.. i'll get rid of extra class

**Pull request merged**